src/Http/Routing/src/Builder/RoutingEndpointConventionBuilderExtensions.cs

```diff
@@ -97,15 +80,8 @@ public static class RoutingEndpointConventionBuilderExtensions
     /// <returns>The <see cref="IEndpointConventionBuilder"/>.</returns>
     public static TBuilder WithMetadata<TBuilder>(this TBuilder builder, params object[] items) where TBuilder : IEndpointConventionBuilder
     {
-        if (builder == null)
-        {
-            throw new ArgumentNullException(nameof(builder));
-        }
-
-        if (items == null)
-        {
-            throw new ArgumentNullException(nameof(items));
-        }
+        ArgumentNullException.ThrowIfNull(builder, nameof(builder));
```

Comment:
Why is nameof needed? Isn't it inferred by default from the expression passed in?

Reply:
Exactly. You are right and I was wrong.
@halter73 Had added MapGroup(https://github.com/dotnet/aspnetcore/pull/41265) pre conditions with param names and I continued the same.

Reply:
It just got fixed